### PR TITLE
preview of questions for profs

### PR DESCRIPTION
**Pull request by ahmedqarmout2**

### CODE REVIEW

File: server/questions.js [hunk 1]
```diff
@@ -126,15 +126,21 @@ var lookupQuestionById = function(questionId, callback) {
...
 
 	lookupQuestionById(questionId, function(err, question){
-		if(err){
-			return callback(err, null);
+		if(err || !question){
+			return callback('error', null);
 		}
 
-		var value = answer===question.answer;
+		var value = (answer === question.answer);
```
Comment (Mohammed-JS-Khan) [inline server/questions.js]: Nice!

File: server/questions.js [hunk 2]
```diff
@@ -126,15 +126,21 @@ var lookupQuestionById = function(questionId, callback) {
...
+
+		if (userType === common.userTypes.ADMIN) {
```
Comment (Mohammed-JS-Khan) [inline server/questions.js:141]: Nice!

File: public/js/question.js [hunk 3]
```diff
@@ -34,17 +33,17 @@ $('#mc_answerform').submit(function(evt) {
...
-            } else if (data.result == 'correct') {
-                swal({
-                    title: 'Correct',
-                    text: 'Congratulations! You gained ' + data.points + ' points!',
-                    type: 'success'
-                }, function () {
-                    window.location.href = '/home';
-                });
-            }
+            swal({
```
Comment (Mohammed-JS-Khan) [inline public/js/question.js:36]: Both these functions are almost identical. But I assume theres no way to combine them? Maybe make the success functions run the same function? Not sure of the implications there though.

**Review by Mohammed-JS-Khan — APPROVED** [review]
Solid PR fam 👍

**Review by mehtadarshan1 — APPROVED** [review]

File: public/js/question.js (hunk 3 above)
Comment (ahmedqarmout2) [reply public/js/question.js:36]: I will try to combine them

File: server/questions.js [hunk 4]
```diff
@@ -126,15 +126,21 @@ var lookupQuestionById = function(questionId, callback) {
  * Update the question object in the database and call the callback function
  * with the result of the comparison and the new question object.
  */
-exports.checkAnswer = function(questionId, userId, answer, callback) {
+exports.checkAnswer = function(questionId, user, answer, callback) {
```
Comment (esteeroramy) [inline server/questions.js]: indentation in this function seems kinda off, just take another look at it :)

**Review by esteeroramy — APPROVED** [review]

File: server/questions.js (hunk 4 above)
Comment (ahmedqarmout2) [reply server/questions.js]: fixed

**Pull request merged**